### ISSUE TITLE
Newer libstartup-notification dependency

### ISSUE THREAD

**Issue by myared** [Airblader/i3#125]
```
checking for LIBSN... no
configure: error: Package requirements (libstartup-notification-1.0) were not met:

No package 'libstartup-notification-1.0' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables LIBSN_CFLAGS
and LIBSN_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

I'm having trouble configuring on Ubuntu 16.10. The issue appears to be that I need libstartup-notification-1.0 whereas I have libstartup-notification-0.12 installed. 

```
libstartup-notification0/yakkety,now 0.12-4build1 amd64 [installed]
  library for program launch feedback (shared library)

libstartup-notification0-dev/yakkety,now 0.12-4build1 amd64 [installed]
  library for program launch feedback (development headers)
```

**Comment by Airblader** [comment]
What do you expect me to do about it? Install the correct dependency.

**Comment by Airblader** [comment]
Actually 0.10 / 0.12 is the correct dependency version, not 1.0. Given that many other people have been able to compile on Ubuntu this feels like it might be a setup issue on your part.

Either way, this is not specific to i3-gaps, so please report upstream in i3. As I state in the i3-gaps wiki, I generally try not to answer further questions about compilation here anyway unless you plan to provide a package.

**Comment by Airblader** [comment]
(To clarify, we do of course support compilation problems upstream)

**Comment by myared** [comment]
Makes sense. I'll keep looking.

**Comment by IniZio** [comment]
Hi @myared  I also got the same issue despite having libstartup-notification 0.12 installed, did you have the solution as for the error?

**Comment by ponelat** [comment]
So I found myself with this issue, and after a bit of bashing my head I eventually tried `sudo ../configure --prefix=/usr --sysconfdir=/etc --disable-sanitizers` and that finally worked. 

I'm not sure if there are safety implications here, so please be advised. But this did get me over that hurdle. 

Hope it helps someone!

**Comment by Airblader** [comment]
@ponelat Not sure what that line would have to do with dependencies, but apart from sudo, that's how I build the i3-gaps package for Arch.

**Comment by ponelat** [comment]
@Airblader yeah, its the sudo part that made a difference... no idea why.
But when I followed the instructions to the letter, it didn't get past `No package 'libstartup-notification-1.0' found`.

I then went off into the deep internet to uncover how to get that package, failed and then just prefxed the `configure` line with sudo. And :tada: it worked.

ie:
#### Original
```sh
cd /path/where/you/want/the/repository

# clone the repository
git clone https://www.github.com/Airblader/i3 i3-gaps
cd i3-gaps

# compile & install
autoreconf --force --install
rm -rf build/
mkdir -p build && cd build/

# Disabling sanitizers is important for release versions!
# The prefix and sysconfdir are, obviously, dependent on the distribution.
../configure --prefix=/usr --sysconfdir=/etc --disable-sanitizers
make
sudo make install
```

#### With _configure_ prefixed with `sudo`
Its likely a permissions thing with my user, but there isn't much magic associated with it.
Linux Mint 18, based on Ubuntu Xenial 16.04 
```sh
cd /path/where/you/want/the/repository

# clone the repository
git clone https://www.github.com/Airblader/i3 i3-gaps
cd i3-gaps

# compile & install
autoreconf --force --install
rm -rf build/
mkdir -p build && cd build/

# Disabling sanitizers is important for release versions!
# The prefix and sysconfdir are, obviously, dependent on the distribution.
sudo ../configure --prefix=/usr --sysconfdir=/etc --disable-sanitizers
make
sudo make install
```

Again, its only if it helps solve someone else's issue. As I found this issue, based on the OP's comment.

**Comment by mikefox10** [comment]
install libstartup-notification0-dev in debian stretch

**Comment by nirvdrum** [comment]
As an additional FYI, I found the reason using `sudo` worked in my case (Ubuntu 19.04) is because my user's `PATH` picked up a different `pkg-config` than the standard system one. The only thing `sudo` did was reset my `PATH` for me. In my case, the alternative `pkg-config` came from a Homebrew/Linuxbrew installation.

I was able to run the configure script as an unprivileged user by setting `PKG_CONFIG=/usr/bin/pkg-config`.
